agent/consul/discoverychain/compile.go

```diff
@@ -736,6 +737,10 @@ func (c *compiler) newTarget(opts structs.DiscoveryTargetOpts) *structs.Discover
 		// Use the same representation for the name. This will NOT be overridden
 		// later.
 		t.Name = t.SNI
+		peer := c.entries.Peers[opts.Peer]
+		if peer != nil && peer.Remote != nil && peer.Remote.Locality != nil {
+			t.Locality = pbpeering.LocalityToStruct(peer.Remote.Locality)
```

Comment:
Could we add this to an existing test so that we know it's populating data correctly?